tests/unit/test_completion.py

```
@@ -95,6 +95,23 @@ async def _empty_stream() -> AsyncIterator[Any]:
     assert mock_ac.call_args.kwargs.get("timeout") == 600
 
 
+@pytest.mark.asyncio
+async def test_acompletion_rejects_timeout_for_unsupported_provider() -> None:
+    # A provider that declares no per-request timeout support rejects a caller-supplied timeout
+    # centrally, before the request is built.
+    provider = AnyLLM.create("openai", api_key="sk-test")
+    provider.TIMEOUT_SUPPORT = "unsupported"
+    with patch.object(provider, "_acompletion", new=AsyncMock()) as mock_ac:
+        with pytest.raises(UnsupportedParameterError, match="timeout"):
```

Ruff (0.16.1)

[warning] Use a single `with` statement with multiple contexts instead of nested `with` statements

(SIM117)